### PR TITLE
GS-2 Created Genre entity and components

### PR DESCRIPTION
**Pull request by JuleaRazvan**

### CODE REVIEW

File: src/main/java/com/project/gamestore/services/GenreService.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                .toList();
+    }
+
+    public GenreDTO getByPublicIdentifier(UUID publicIdentifier) {
+        Genre foundGenre = genreRepository.findByPublicIdentifier(publicIdentifier).orElseThrow();
+        return genreMapper.mapEntityToDTO(foundGenre);
+    }
+
+    public GenreDTO update(GenreDTO genreDTO, UUID publicIdentifier) {
+        Genre updatGenre = genreRepository.findByPublicIdentifier(publicIdentifier).orElseThrow();
```
Comment (RaresAlexandruAlic) [inline src/main/java/com/project/gamestore/services/GenreService.java]: You ate a letter here maybe `updateGenre`

File: src/main/resources/liquibase/changes-1.0.xml [hunk 2]
```diff
@@ -28,4 +28,30 @@
...
       </createTable>
    </changeSet>
+
+   <changeSet id="gs-2.1" author="rjulea">
+      <createTable tableName="genres">
+         <column name="id" type="bigint" autoIncrement="true">
+            <constraints primaryKey="true" nullable="false" uniqueConstraintName="users_pk"/>
+         </column>
+         <column name="public_identifier" type="${uuid.type}">
+            <constraints nullable="false" unique="true" uniqueConstraintName="gener_public_identifier_uk"/>
```
Comment (RaresAlexandruAlic) [inline src/main/resources/liquibase/changes-1.0.xml]: please change the `uniqueConstraintName ` to spell genre correctly

File: src/main/resources/liquibase/changes-1.0.xml [hunk 3]
```diff
@@ -28,4 +28,30 @@
...
+         <column name="last_updated_at" type="${timestamp.type}">
+            <constraints nullable="false"/>
+         </column>   
+         <column name="name" type="varchar(50)">
+            <constraints nullable="false"/>
+         </column>
+         <column name="popularity" type="bigint">
+            <constraints nullable="false"/>
+         </column>
+         <column name="description" type="varchar(50)">
```
Comment (RaresAlexandruAlic) [inline src/main/resources/liquibase/changes-1.0.xml]: Maybe increase the dimension for the description if you want. You might have long sentences for some descriptions that might not fit in 50 letters.

**Review by RaresAlexandruAlic — APPROVED** [review]
Looks good. Found just some misspelled variables and I've also left a suggestion on your db entity changeset. Other than that great work!

File: src/main/java/com/project/gamestore/services/GenreService.java (hunk 1 above)
Comment (JuleaRazvan) [reply src/main/java/com/project/gamestore/services/GenreService.java]: Resolved

**Pull request merged**